### PR TITLE
[REVIEW] fixing arrow compilation

### PR DESCRIPTION
**Pull request by Christian8491**
This PR also  fix issues related to new parquet API in cudf.

### CODE REVIEW

File: io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ return arrow::Status::OK();
 		//		results.GetResult().GetBody().read((char*)(out->get()->mutable_data()), bytesRead);
 
 		std::shared_ptr<arrow::ResizableBuffer> buffer;
-		AllocateResizableBuffer(arrow::default_memory_pool(), nbytes, &buffer);
+		AllocateResizableBuffer(nbytes, arrow::default_memory_pool());
```
Comment (aucahuasi) [inline io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp:181]: Should be something like this:
`buffer = AllocateResizableBuffer(nbytes, arrow::default_memory_pool());`
We need to make sure we are allocating the buffer

Comment (Christian8491) [reply io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp]: Sure. thanks for the advice.

File: conda/recipes/blazingsql/meta.yaml [hunk 2]
```diff
@@ -28,6 +28,7 @@ requirements:
         - python
         - pip
         - setuptools
+        - spdlog >=1.7.0,<1.8.0
```
Comment (aucahuasi) [inline conda/recipes/blazingsql/meta.yaml:31]: Why we need to declare spdlog on here? Is already declared in the rmm package
https://github.com/rapidsai/rmm/blob/branch-0.16/conda/recipes/librmm/meta.yaml#L31
We should avoid to include here transitive dependencies cc @mario21ic

File: io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp [hunk 3]
```diff
@@ -177,18 +178,18 @@ return arrow::Status::OK();
 		//}
 		//		results.GetResult().GetBody().read((char*)(out->get()->mutable_data()), bytesRead);
 
-		std::shared_ptr<arrow::ResizableBuffer> buffer;
-		AllocateResizableBuffer(arrow::default_memory_pool(), nbytes, &buffer);
+		arrow::Result<std::unique_ptr<arrow::ResizableBuffer>> buffer = AllocateResizableBuffer(nbytes, arrow::default_memory_pool());
 
-		results.read((char *) (buffer->mutable_data()), bytesRead);
+		results.read((char *) (buffer.ValueOrDie()->mutable_data()), bytesRead);
```
Comment (aucahuasi) [inline io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp:183]: How we handle the "die" part? For example that arrow API says
`buffer.ValueOrDie()`
I think in most cases we will get the buffer but there will be be some cases where we get a "die" object/buffer ... how do we handle that? maybe we need to raise with low exception here cc @felipeblazing

File: conda/recipes/blazingsql/meta.yaml (hunk 2 above)
Comment (williamBlazing) [reply conda/recipes/blazingsql/meta.yaml:31]: without it, its picking the latest spdlog 1.8.0 which we dont want.Not sure why the rmm one is not being used

File: io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp (hunk 3 above)
Comment (williamBlazing) [reply io/src/FileSystem/private/GoogleCloudStorageReadableFile.cpp:183]: The exception handling of all of this code is in desperate need of cleanup. It has been outdated for a long time. I think that needs to be addressed by a different PR

**Pull request merged**